src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/PolicyFunctions.kt

```diff
@@ -25,42 +32,77 @@ fun <T : ILicenceCondition> conditionChanges(current: List<T>, other: List<T>) =
   amendedConditions(current, other)
 )
 
-/*
- *  For each condition, create List item of custom type
- *  Add new list of HashMap<placeholder, variable> to condition list
- *  Iterate through placeholders and add to HashMap with corresponding date field ({} from tpl)
- */
-fun <T : ILicenceCondition> getPolicyPlaceholders(conditions: List<T>): HashMap<String, HashMap<String, String>> {
-  val placeholderPattern = Regex("(?<=\\[)(.*?)(?=])")
-  val variablePattern = Regex("(?<=\\{)(.*?)(?=})")
-  val results = HashMap<String, HashMap<String, String>>()
+fun <T : Any> licencePolicyChanges(
+  presentConditions: List<AdditionalCondition>,
+  previousConditions: List<IAdditionalCondition<T>>,
+  currentConditions: List<IAdditionalCondition<T>>,
+  allReplacements: List<Replacements>
+): List<LicenceConditionChanges> {
+  val conditionsToCheck = currentConditions.filter { current ->
+    presentConditions.any { present -> current.code == present.code }
+  }
+  val changes = conditionChanges(conditionsToCheck, previousConditions, allReplacements)
 
-  conditions.filter { it.requiresInput }.forEach {
-    val placeholders = placeholderPattern.findAll(it.text).iterator()
-    val variables = variablePattern.findAll(it.tpl ?: "").toSet().iterator()
-    val keyValuePairs = HashMap<String, String>()
+  return changes.map { change -> change.copy(sequence = presentConditions.find { condition -> condition.code == change.code }?.sequence) }
```

Comment:
this is kinda gnarly - re-stitch back licence specific info about the sequence.

But the tradeoff is that `conditionChanges` is simpler as we don't have to pass through licence specific data